firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/TimestampSerializer.kt

```diff
@@ -54,13 +54,13 @@ object DoubleAsTimestampSerializer : SpecialValueSerializer<Double>(
     toNativeValue = { value ->
         when(value) {
             serverTimestamp -> FieldValue.serverTimestamp.nativeValue
-            else -> Timestamp.fromMilliseconds(value)
+            else -> Timestamp.fromMilliseconds(value.toLong())
```

Comment:
floating part causes all kinds of conversion noise. it's better to be precise to milliseconds than precise to seconds